File: trees/basic.java
import java.util.*;
class Node
{
	int data;
	Node left, right;
	public Node(int data)
	{
		this.data = data;
		left = null;
		right = null;
	}
}

class basic
{
	Node root;

	public basic()
	{
		root = null;
	}

	public static void main(String[] args)
	{
		basic tree = new basic();

		tree.root = new Node(10);
		tree.root.left = new Node(20);
		tree.root.right = new Node(30);
		tree.root.left.left = new Node(40);
	}
}

File: trees/basic.c
#include<stdio.h>
#include<stdlib.h>
struct node
{
	int data;
	struct node *left;
	struct node *right;
};
struct node * NewNode(int data)
{
	struct node *n = (struct node*)malloc(sizeof(struct node));
	n->data = data;
	n->left = NULL;
	n->right = NULL;
	return n;
}
int main()
{
	struct node *root = NewNode(10);
	root->left = NewNode(20);
	root->right = NewNode(30);
	return 0;
}